cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -118,26 +118,23 @@ public static KafkaCluster fromConfigMap(ConfigMap kafkaClusterCm) {
                 Labels.fromResource(kafkaClusterCm));
 
         Map<String, String> data = kafkaClusterCm.getData();
-        kafka.setReplicas(Integer.parseInt(data.getOrDefault(KEY_REPLICAS, String.valueOf(DEFAULT_REPLICAS))));
-        kafka.setImage(data.getOrDefault(KEY_IMAGE, DEFAULT_IMAGE));
-        kafka.setHealthCheckInitialDelay(Integer.parseInt(data.getOrDefault(KEY_HEALTHCHECK_DELAY, String.valueOf(DEFAULT_HEALTHCHECK_DELAY))));
-        kafka.setHealthCheckTimeout(Integer.parseInt(data.getOrDefault(KEY_HEALTHCHECK_TIMEOUT, String.valueOf(DEFAULT_HEALTHCHECK_TIMEOUT))));
+        kafka.setReplicas(getInteger(data, KEY_REPLICAS, DEFAULT_REPLICAS));
+        kafka.setImage(getNonemptyString(data, KEY_IMAGE, DEFAULT_IMAGE));
+        kafka.setHealthCheckInitialDelay(getInteger(data, KEY_HEALTHCHECK_DELAY, DEFAULT_HEALTHCHECK_DELAY));
+        kafka.setHealthCheckTimeout(getInteger(data, KEY_HEALTHCHECK_TIMEOUT, DEFAULT_HEALTHCHECK_TIMEOUT));
 
-        kafka.setZookeeperConnect(data.getOrDefault(ENV_VAR_KAFKA_ZOOKEEPER_CONNECT, kafkaClusterCm.getMetadata().getName() + "-zookeeper:2181"));
+        kafka.setZookeeperConnect(kafkaClusterCm.getMetadata().getName() + "-zookeeper:2181");
 
         String metricsConfig = data.get(KEY_METRICS_CONFIG);
         kafka.setMetricsEnabled(metricsConfig != null);
         if (kafka.isMetricsEnabled()) {
             kafka.setMetricsConfig(new JsonObject(metricsConfig));
         }
 
-        String storageConfig = data.get(KEY_STORAGE);
-        kafka.setStorage(Storage.fromJson(new JsonObject(storageConfig)));
+        kafka.setStorage(getStorage(data, KEY_STORAGE));
 
-        String kafkaConfig = data.get(KEY_KAFKA_CONFIG);
-        if (kafkaConfig != null) {
-            kafka.setConfiguration(new KafkaConfiguration(new JsonObject(kafkaConfig)));
-        }
+        JsonObject kafkaConfig = getConfig(data, KEY_KAFKA_CONFIG);
+        kafka.setConfiguration(new KafkaConfiguration(kafkaConfig));
```

Comment:
If we are changing the configuration from using `null` to being empty but not null, we should maybe remove the unnecessary `null` check here: https://github.com/strimzi/strimzi/blob/23a12d8452666963ab5cbdedd2fb2b5102080670/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java#L304